Cubical/HITs/Hopf.agda

```diff
@@ -18,16 +38,286 @@ HopfS² (surf i j) = Glue S¹ (λ { (i = i0) → _ , idEquiv S¹
                                ; (j = i1) → _ , _ , rotIsEquiv (loop i) } )
 
 -- Hopf fibration using more direct definition of the rot equivalence
+-- TODO : prove that it is equivalent to HopfSuspS¹
 HopfS²' : S² → Set
 HopfS²' base = S¹
 HopfS²' (surf i j) = Glue S¹ (λ { (i = i0) → _ , rotLoopEquiv i0
                                 ; (i = i1) → _ , rotLoopEquiv i0
                                 ; (j = i0) → _ , rotLoopEquiv i0
                                 ; (j = i1) → _ , rotLoopEquiv i } )
 
--- Hopf fibration using suspension of S¹
-HopfSuspS¹ : SuspS¹ → Set
-HopfSuspS¹ north = S¹
-HopfSuspS¹ south = S¹
-HopfSuspS¹ (merid x i) = ua (_ , rotIsEquiv x) i
+-- Total space of the fibration
+TotalSpace : Set
+TotalSpace = Σ SuspS¹ HopfSuspS¹
+
+-- Forward direction
+filler-1 : I → (j : I) → (y : S¹) → Glue S¹ (Border y j) → join S¹ S¹
+filler-1 i j y x = hfill (λ t → λ { (j = i0) → inl (rotInv-1 x y t)
+                                  ; (j = i1) → inr x })
+                         (inc (push ((unglue (j ∨ ~ j) x) * inv y) (unglue (j ∨ ~ j) x) j)) i
+
+Hopf→Join : TotalSpace → join S¹ S¹
+Hopf→Join (north , x) = inl x
+Hopf→Join (south , x) = inr x
+Hopf→Join (merid y j , x) = filler-1 i1 j y x
+
+-- Backward direction
+Join→Hopf : join S¹ S¹ → TotalSpace
+Join→Hopf (inl x) = (north , x)
+Join→Hopf (inr x) = (south , x)
+Join→Hopf (push y x j) =
+  (merid (inv y * x) j
+  , glue (λ { (j = i0) → y ; (j = i1) → x }) (rotInv-2 x y j))
+
+-- Now for the homotopies, we will need to fill squares indexed by x y : S¹ with value in S¹
+-- Some will be extremeley tough, but happen to be easy when x = y = base
+-- therefore, we fill them for x = y = base and then use the connectedness of S¹ × S¹ and
+-- the discreteness of ΩS¹ to get general fillers.
+
+-- To proceed with that strategy, we first need a lemma :
+-- the sections of the trivial fibration λ (_ : S¹) (_ : S¹) → Int are constant
+
+-- this should be generalized to a constant fibration over a connected space with
+-- discrete fiber
+fibInt : S¹ → S¹ → Set
+fibInt _ _ = Int
+
+-- TODO: this should be moved to a more general place, like HLevel.agda
```

Comment:
The names for the following few results are not so good. Instead of `Set` use `isSet`.  Also maybe this should be generalized (so that it says something about connected types instead)?